crates/oxc_linter/src/rules/eslint/func_names.rs

```diff
@@ -455,7 +455,30 @@ impl Rule for FuncNames {
                     |fixer| {
                         guess_function_name(ctx, parent_node.id()).map_or_else(
                             || fixer.noop(),
-                            |name| fixer.insert_text_after(&replace_span, format!(" {name}")),
+                            |name| {
+                                // if this name shadows a variable in the outer scope **and** that name is referenced
+                                // inside the function body, it is unsafe to add a name to this function
+                                if ctx
+                                    .scopes()
+                                    .find_binding(func.scope_id.get().unwrap(), &name)
+                                    .map_or(false, |shadowed_var| {
+                                        ctx.semantic().symbol_references(shadowed_var).any(
+                                            |reference| {
+                                                func.span.contains_inclusive(
```

Comment:
let me know if there's a better way than checking the span, but this felt easiest rather than collecting up all child scope ids of the function, and checking whether the reference is inside one of those ids